src/main/java/net/neoforged/neoforge/flag/Flag.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.flag;
+
+import com.google.common.collect.Maps;
+import com.mojang.serialization.Codec;
+import io.netty.buffer.ByteBuf;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+import net.minecraft.network.codec.StreamCodec;
+import net.minecraft.resources.ResourceLocation;
+
+/**
+ * Class describing a specific flag.
+ */
+public final class Flag {
+    public static final Codec<Flag> CODEC = ResourceLocation.CODEC.xmap(Flag::of, flag -> flag.identifier);
+    public static final StreamCodec<ByteBuf, Flag> STREAM_CODEC = ResourceLocation.STREAM_CODEC.map(Flag::of, flag -> flag.identifier);
+
+    private static final Map<ResourceLocation, Flag> FLAGS = Maps.newConcurrentMap();
+    private static final Collection<Flag> FLAGS_VIEW = Collections.unmodifiableCollection(FLAGS.values());
+
+    private final ResourceLocation identifier;
+
+    private Flag(ResourceLocation identifier) {
+        this.identifier = identifier;
+    }
+
+    /**
+     * @return The owning mod id/namespace for this {@link Flag flag}.
+     */
+    public String namespace() {
+        return identifier.getNamespace();
+    }
+
+    /**
+     * @return The unique identifier for this {@link Flag flag}.
+     * @apiNote This identifier does not need to be globally unique, only unique to the owning mod.
+     */
+    public String identifier() {
+        return identifier.getPath();
+    }
+
+    public String toStringShort() {
+        return identifier.toString();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj.getClass() == Flag.class)
+            return identifier.equals(((Flag) obj).identifier);
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return identifier.hashCode();
+    }
+
+    @Override
+    public String toString() {
+        return "Flag{" + identifier + '}';
+    }
+
+    /**
+     * Constructs a new {@link Flag} from the given {@code namespace} and {@code identifier}.
+     *
+     * @param namespace  The owning mods id/namespace.
+     * @param identifier A mod specific unique identifier.
+     * @return The newly constructed flag.
+     */
+    public static Flag of(String namespace, String identifier) {
```

Comment:
Not a fan of this overload. Most mods have something akin to a `MyMod.RL(...)` method, so this is typically unnecessary.

Reply:
While this is true, should we be relying on mods to have said helpers in their mods?
I also personally find `Flag.of(MyMod.RL("id"))` or even `Flag.of(RL.fromNamespace(MyMod.ID, "id"))` to be quite ugly code and prefer `Flag.of(MyMod.ID, "id")`

But as mentioned before the RL here could be dropped entirely, its just used for the handy namespaced string helpers it provides, not rly using the whole RL api.